### PR TITLE
Docs: Update plugin.json doc as version is not required

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:

This PR updates the plugin.json doc so the version property of dependencies as it is neither required nor should it use semver ranges. See the accompanying plugin.schema [PR here](https://github.com/grafana/grafana/pull/90142).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by tolzhabayev — APPROVED** [review]

**Review by leventebalogh — APPROVED** [review]

**Review by xnyo — APPROVED** [review]
LGTM!

**Review by wbrowne — APPROVED** [review]

**Review by josmperez — APPROVED** [review]
LGTM.

File: docusaurus/docs/reference/metadata.md [hunk 1]
```diff
@@ -65,12 +65,12 @@ Plugin dependency. Used to display information about plugin dependencies in the
...
+| Property  | Type   | Required | Description                                                                                                                                           |
+| --------- | ------ | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `id`      | string | **Yes**  |                                                                                                                                                       |
+| `name`    | string | **Yes**  |                                                                                                                                                       |
+| `type`    | string | **Yes**  | Possible values are: `app`, `datasource`, `panel`.                                                                                                    |
+| `version` | string | **No**   | We suggest using [SemVer](https://semver.org/) as your plugin versioning scheme. This property should use a specific, pinned version and not a range. |
```
Comment (marefr) [inline docusaurus/docs/reference/metadata.md:73]: Maybe add something along the lines "if no version specified, the latest supported version will be used"?

Comment (jackw) [reply docusaurus/docs/reference/metadata.md]: Ah thanks @marefr . Added in c2858fbd4dbc0a8324b0ace4a871afe613081308 👍

Comment (jackw) [reply docusaurus/docs/reference/metadata.md]: It's not really up to the developer either as I'm pretty sure all our tooling and some checks within codebases is geared around Semver and if they were to use anything else (such as date of release) things would go wrong somewhere. How about I reword it to:

```
This property should use a specific, pinned [SemVer](https://semver.org/) version and not a range. If no version is specified, it will resolve to the latest supported version.
```

and we update the `info.version` property blurb to read something like:

```diff
-Project version of this commit, e.g. `6.7.x`
+[SemVer](https://semver.org/) version of the plugin, e.g. `6.7.1`.
```

Comment (sympatheticmoose) [reply docusaurus/docs/reference/metadata.md]: LGTM 🚀
